documentation/quarkus/guide-api-first.asciidoc

```diff
@@ -0,0 +1,153 @@
+:toc: macro
```

Comment:
This is a general guide about API first and not only Quarkus specific, right? So I think we should move it from the quarkus subfolder to the documentation folder.